warehouse/legacy/api/pypi.py

```diff
@@ -194,7 +195,8 @@ def _validate_project_url(value):
     if not url:
         raise wtforms.validators.ValidationError("Must have an URL.")
 
-    # TODO: Actually validate that the URL is a valid URL.
+    if not rfc3986.is_valid_uri(url, require_scheme=True):
```

Comment:
This should probably include a check to make sure that the scheme we do get is either http or https. This will also accept things like ``ssh://foobar`` which I think is not something that we want to actually do here. I think it shouldn't be super hard to adjust this to check the scheme too though.